### PR TITLE
Node 18+ updates

### PR DESCRIPTION
**Pull request by fajar-apri-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #63, #61, #62, #64, #65, #66, #68, #69, #70

## Summary:

...

## Type of change:

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- JetStream Design System Team

### CODE REVIEW

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: web-test-runner.config.mjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  nodeResolve: true,
+  coverageConfig: {
+    threshold: {
+      statements: 80,
+      branches: 80,
+      functions: 80,
+      lines: 80
+    }
+  }
+};
```
[the remark this answers is not in the document]
Comment (fajar-apri-alaska) [reply web-test-runner.config.mjs:12]: Will do, thanks for the reminder.

File: rollup.config.mjs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    nodeResolve(),
+    !production &&
+      serve({
+        open: true,
+        openPage: '/docs/'
+      })
+  ]
+};
+
+export default [modernConfig];
```
[the remark this answers is not in the document]
Comment (fajar-apri-alaska) [reply rollup.config.mjs:24]: Will do, thanks for the reminder.

**Comment by fajar-apri-alaska** [comment]
Hi @blackfalcon, the "Test and publish" check is failing again, I did npm install in latest node 18.
Any other configuration that is missing from my side?

**Comment by blackfalcon** [comment]
> Hi @blackfalcon, the "Test and publish" check is failing again, I did npm install in latest node 18. Any other configuration that is missing from my side?

When running into these issues, please review the error logs in the actions UI. This is the issue. 
https://github.com/AlaskaAirlines/auro-header/actions/runs/4719720581/jobs/8370882711#step:4:17

This is the fix
https://github.com/AlaskaAirlines/auro-header/actions/runs/4719720581/jobs/8370882711#step:4:29

You should have also seen this in your localhost logs.

**Comment by blackfalcon** [comment]
@fajar-apri-alaska I see that you are adding the links to the issues in progress in the PR message itself, please also add these references in the DEVELOPMENT section of the PR as well. 

Screen Shot 2023-04-17 at 7 45 50 AM

**Comment by blackfalcon** [comment]
@fajar-apri-alaska it's important that you add the correct information into the body of the commit message. The following commit is missing the `BREAKING CHANGE` message that it requires. 

Screen Shot 2023-04-17 at 10 00 02 AM

Please see this example from auro-badge. 
https://github.com/AlaskaAirlines/auro-badge/pull/45/commits/2812f8d70c4a26d6f552bb880c07603b7ca2942d

File: src/auro-header.js [hunk 3]
```diff
@@ -4,7 +4,7 @@
 // ---------------------------------------------------------------------
 
 import { LitElement, html } from "lit-element";
```
Comment (blackfalcon) [inline src/auro-header.js:6]: This was missed. 

```
import { LitElement, html } from "lit";
```

File: package.json [hunk 4]
```diff
@@ -22,7 +22,8 @@
   "license": "Apache-2.0",
   "dependencies": {
     "chalk": "^4.1.2",
-    "lit": "^2.7.2"
+    "lit": "^2.7.2",
+    "typescript": "^5.0.4"
```
Comment (blackfalcon) [inline package.json:26]: This is not a root dependency. Update to be a dev dependency.

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "declaration": true,
+    // This compiler run should
+    // only output d.ts files
+    "emitDeclarationOnly": true,
+    "outDir": "dist",
+    // go to js file when using IDE functions like
+    // "Go to Definition" in VSCode
+    "declarationMap": true
+  }
+}
```
Comment (blackfalcon) [inline tsconfig.json:18]: Always add the extra line break.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
+  - type: dropdown
+    id: repro
+    attributes:
+      label: Reproducing the error
+      description: Are you able to reproduce this issue on the [Auro docsite](https://auro.alaskaair.com/)?
+      options:
+        - Yes I can
+        - No I can't
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:35]: Please reference this commit

https://github.com/AlaskaAirlines/auro-badge/commit/7da226006c43fdd0ad26c49abd84612d11b2b4aa

File: .github/ISSUE_TEMPLATE/general-support.yml [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
++25,492 −16,077 
+ Conversation 1
+ Commits 14
+ Checks 2
+ Files changed 40
+ Merged
+Node 18+ updates
+#45
+Show file tree Hide file tree
+File filter
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/general-support.yml:35]: This whole file is incorrect? See reference https://github.com/AlaskaAirlines/auro-badge/pull/45/commits/a626ef293c05e6abaf4e222b3e3f40bd24ed7b15#diff-f98eee0578914954ee0aee3193f65adbf0f477f7ee2a3edf11570d25777bed08

File: src/style-fixed.scss [hunk 8]
```diff
@@ -1,4 +1,4 @@
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties';
+@import './../node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties';
```
Comment (blackfalcon) [inline src/style-fixed.scss:1]: This file should be deleted.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
There are missing updates. Please reference the following updates. 

- https://github.com/AlaskaAirlines/auro-badge/pull/45/files#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5b
- https://github.com/AlaskaAirlines/auro-badge/pull/45/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7
- https://github.com/AlaskaAirlines/auro-badge/pull/45/files#diff-2bdae0a46b80ac55c81e2b572d97d11454b86cad730cf0370164e80296fbc0ba
- https://github.com/AlaskaAirlines/auro-badge/pull/45/files#diff-cac4e090272551ada3915212866aa18637b83dac40952859e496b0af91e20e40
- https://github.com/AlaskaAirlines/auro-badge/pull/45/files#diff-43207385fe61862722b57bd45d8bef3c30b504feb8b7342a02088fee2cda3c46

Remove this file and its related resources 
https://github.com/AlaskaAirlines/auro-header/blob/fajarapri/node18update/scripts/polyfills.js

File: .github/ISSUE_TEMPLATE/general-support.yml (hunk 7 above)
Comment (fajar-apri-alaska) [reply .github/ISSUE_TEMPLATE/general-support.yml:35]: did I just do this? Let me check, sorry.

**Review by blackfalcon — DISMISSED** [review]

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**